### ISSUE TITLE
Color picker section spills out of designated space when too many custom colors are added

### ISSUE THREAD

**Issue by acornestean** [mozilla/FirefoxColor#934]
Preconditions:
1. Go to about:config and change the “xpinstall.signatures.required” pref to “false”
2. Access https://color.stage.mozaws.net/
3. Click on the “Advanced colors” tab
4. Click on “Firefox Color Extension” link from the right side panel
5. On the newly opened page, click on “Unsigned add-on for color.stage.mozaws.net “
6. Follow the instructions on the doorhangers and install the Firefox Color add-on

Steps to reproduce:
1. Access https://color.stage.mozaws.net/
2. Click on the “Advanced colors” tab
3. Set a different color for each property
NOTE: Notice how each time a new/custom color is set for a property, a color square is added at the bottom of the Color picker section
4. Observe that when new/custom colors are set for each property, the Color picker section enlarges vertically and spills out of the designated space

Expected results:
The Color picker section should retain its dimensions when new/custom colors are added for each property.

Actual results:
The Color picker section enlarges vertically and spills out when new/custom colors are added for each property.

Affected versions:
- Nightly 79.0b1/20200629224419
- Beta Developer Edition 80.0a1/20200629154604

Affected OSes:
- Windows 10 64-bit
- Ubuntu 16.04 LTS

Nightly screenshot:
![Nightly spill](https://user-images.githubusercontent.com/50236075/86121341-867bfc80-bade-11ea-8c2e-1ec70bac2736.png)

Beta Developer Edition screenshot:
![Beta spill](https://user-images.githubusercontent.com/50236075/86121372-94318200-bade-11ea-8195-85c8977c99a4.png)

**Comment by acornestean** [comment]
The issue also occurs on Unbranded Release (77.0.1/20200602222727).

**Comment by acornestean** [comment]
The issue also occurs on macOS Catalina 10.15 as well.

**Comment by andreicristianpetcu** [comment]
@Rob--W should I increase the height of the container?

**Comment by Rob--W** [comment]
I didn't look at the code, but I wouldn't be surprised that the color picker has a flexible maximum height, but that it doesn't account for the vertical space used by the radio buttons + labels above it.

If that is the case, the correct fix would be to update the stylesheet to account for it.

Have you tried to look into the reason for the overflowing color picker?

**Comment by andreicristianpetcu** [comment]
I only debugged the other one https://github.com/mozilla/FirefoxColor/issues/933
Will debug this in the following days.

**Comment by andreicristianpetcu** [comment]
Here is a URL to test http://localhost:8080/?theme=XQAAAAJkBAAAAAAAAABBKYhm849SCicxcUhA3DJozHnOMuprjJvk-_nNDqSKHtgQaNQFl-kzjFk5azS5uYzbkxY5IUgg66HUg57HiM-NZ7wq2LhoN8zuKpmn9s9r5gjodn8vAo5JpaGEGeCaVkqBjA06ppfhtPbW8Tu9xx65L74OzIifZD80wrQQ4iv6IPy6t32uDBODpL0QRh_Q4CNLKqkxuzock9gUoLVszpI0UjSH5W9Pg9XK0vW76a72jLCMlQMBxTGyyfi-j7SwSDHhfY5m1xn05h9C2n32sxU0i1St8NydnX11RG15lgrQ3UtBEqCqTlU5Y3K9XMy6orSkHen4Z5tLVqiZC5s51DtW4SmtPJBNFavO0bY5MUvZa6_-1svzsSB_vLUH3QdQyU06qzOt2I_m_UwgL8F_ajsO6tffIsLEkg4gFERNQnD9-egvGHjZ0hc5Nr53uW-exzBFKcPilxFG-TmtosWhJijotctyLiYRRbMYNP2MJKr7nUsECinKS9AHfZbzR5U1oZMw8LRMykBz2O7n6ke6nmPGNv7IeB1Af_F41mUkeAd336UqZqEXk_HKtOIsoPKZ7WlBQw9WH_Edn7ygbL_es2jWEJqEVfJD6jE0Wg-SMO6IZCY--nVW1Ek2opWHkjKMXYwpsNPDHOIPJwqgnfLjf0NvJK4zasq1QU_qsJgUhkEDxTKlYWrgsDfXxVnCMrs-iG_8c3nJ

**Comment by andreicristianpetcu** [comment]
.theme-builder__content is the one that sets the height to a fixed one. It's 400px. It looks better with 445px but I'm trying a better fix.

![image](https://user-images.githubusercontent.com/922368/86969338-4bc93280-c176-11ea-8d37-83fbcb34427d.png)

**Comment by andreicristianpetcu** [comment]
Also the picker has all the style inline.
![image](https://user-images.githubusercontent.com/922368/86969504-93e85500-c176-11ea-98b9-f95bc8ca0c16.png)

**Comment by andreicristianpetcu** [comment]
This is my best fix 
![image](https://user-images.githubusercontent.com/922368/86971574-132b5800-c17a-11ea-9216-f0bbac167076.png)

**Comment by Rob--W** [comment]
This looks bad on the standard "Custom colors" tab, because of the unbalanced spacing around the `.storage-space-information` element.

Could you limit the height change to `#theme-builder-tab-content-advanced-colors` only? And while you are at it, please add some spacing above the storage-space-information, so that it doesn't look like it's sticked to the list of advanced colors.

And let's shorten the "use Firefox's default style for this color" label so that it doesn't wrap to two lines. How about "use Firefox's default style"

**Comment by andreicristianpetcu** [comment]
I'm not sure I understand what you want. I set the CSS to this (no other changes):
```
#theme-builder-tab-content-advanced-colors {
  height: 445px;
}
```
And this is how they look
![image](https://user-images.githubusercontent.com/922368/86981945-a8d1e200-c190-11ea-8e06-414f2fab366c.png)
![image](https://user-images.githubusercontent.com/922368/86981951-aec7c300-c190-11ea-8ecb-f4227b966ade.png)

The tabs bar jumps up and down.

**Comment by andreicristianpetcu** [comment]
I only changed the .storage-space-information spacing and the text since I don't understand the request about #theme-builder-tab-content-advanced-colors.
You can edit this yourself if it's faster.
This is how they look now.
![image](https://user-images.githubusercontent.com/922368/86982287-8f7d6580-c191-11ea-839c-8ffd2bc47ba8.png)
![image](https://user-images.githubusercontent.com/922368/86982322-a1f79f00-c191-11ea-8bad-8bd295b8c2c0.png)

![image](https://user-images.githubusercontent.com/922368/86982296-94dab000-c191-11ea-84af-185d87bbf4bc.png)

**Comment by acornestean** [comment]
Verified the fix on the latest Nightly (80.0a1/20200716094032), Beta Dev Edition (79.0b8/20200714212005) and Unbranded Release (78.0.2/20200708170202) under Windows 10 Pro 64-bit and Ubuntu 16.04 LTS.

The Color picker section no longer spills out when adding custom colors to all the properties of the “Advanced color” tab, confirming the fix.
